opm/material/densead/Evaluation.hpp

```diff
@@ -168,15 +186,15 @@ class Evaluation
     {
         // while the values are multiplied, the derivatives follow the product rule,
         // i.e., (u*v)' = (v'u + u'v).
-        const ValueType& u = this->value_;
-        const ValueType& v = other.value_;
-        for (unsigned varIdx = 0; varIdx < size; ++varIdx) {
-            const ValueType& uPrime = this->derivatives_[varIdx];
-            const ValueType& vPrime = other.derivatives_[varIdx];
+        ValueType& u = data_[ valuepos_ ];
```

Comment:
`const ValueType&` is sufficient here? bear in mind that copying value type objects can be expensive in the nested case. (for the usual case, i.e. primitive floating point values there should be no difference.)